test/poseidon.test.ts

```diff
@@ -133,5 +134,72 @@ describe("poseidon test", function () {
         assert.equal(jsOut, "11882816200654282475720830292386643970958445617880627439994635298904836126497", "not equal");
         //assert(poseidon.F.eq(poseidon.F.e("0x11882816200654282475720830292386643970958445617880627439994635298904836126497"), jsOut));
     });
+
+    it("Test poseidon utils using hash for different size inputs", async () => {
+        const generateTemplate = (size) => {
+            const template = `
+pragma circom 2.0.0;
+include "../../circuits/lib/utils/poseidon.util.circom";
+
+template PoseidonTest() {
+    signal input in[{{n}}];
+    signal output out;
+
+    component h = PoseidonUtil({{n}});
+    for(var i = 0; i < {{n}}; i++) {
+        h.in[i] <== in[i];
+    }
+
+    out <== h.out;
+}
```

Comment:
Good :) And such inline test circuits generation is something that can be extended to other tests in the future.